elements/dash-tabs/dash-tabs.js

```diff
@@ -0,0 +1,293 @@
+      newTab.tabIndex = 0;
+      newTab.focus();
+    }
+
+    /**
+     * `_onKeyDown` handles key presses inside the tab panel.
+     */
+    _onKeyDown(event) {
+      // If the keypress did not originate from a tab element itself,
+      // it was a keypress inside the a panel or on empty space. Nothing to do.
+      if (event.target.getAttribute('role') !== 'tab') return;
+      // Don’t handle modifier shortcuts typically used by assistive technology.
+      if (event.altKey) return;
+
+      // The switch-case will determine which tab should be marked as active
+      // depending on the key that was pressed.
+      let newTab;
+      switch (event.keyCode) {
+        // When up or left is pressed, the previous tab is selected.
+        case KEYCODE.LEFT:
+        case KEYCODE.UP:
+          newTab = this._prevTab();
+          break;
+
+        // When right or down is pressed, it’s the next tab.
+        case KEYCODE.RIGHT:
+        case KEYCODE.DOWN:
+          newTab = this._nextTab();
+          break;
+
+        // The home button always selects the first tab.
+        case KEYCODE.HOME:
+          newTab = this._allTabs()[0];
+          break;
+
+        // And end always selects the last one.
+        case KEYCODE.END:
+          const tabs = this._allTabs();
+          newTab = tabs[tabs.length - 1];
+          break;
+        // Any other key press is ignored and passed back to the browser.
+        default:
+          return;
+      }
+
+      // The browser might have some native functionality bound to the arrow
+      // keys, home or end. The element calls `preventDefault` to prevent the
+      // browser from taking any actions.
+      event.preventDefault();
+      // Select the new tab, that has been determined in the switch-case.
+      this._selectTab(newTab);
+    }
+
+    /**
+     * `_onClick` handles clicks inside the tab panel.
+     */
+    _onClick(event) {
+      // If the click was not targeted on a tab element itself,
+      // it was a click inside the a panel or on empty space. Nothing to do.
+      if (event.target.getAttribute('role') !== 'tab') return;
+      // If it was on a tab element, though, select that tab.
+      this._selectTab(event.target);
+    }
+  }
+  window.customElements.define('dash-tabs', DashTabs);
+
+  /**
+   * `DashTab` is a tab for a `<dash-tabs>` tab panel. `<dash-tab>` should
+   * always be used with `role=heading` in the markup so that the semantics
+   * remain useable when JavaScript is failing.
+   *
+   * A `<dash-tab>` declares which `<dash-tabpanel>` it belongs to by using
+   * that panel’s ID as the value for the `aria-controls` attribute.
+   */
+  class DashTab extends HTMLElement {
+    constructor() {
+      super();
+    }
+
+    connectedCallback() {
+      // If this is executed, JavaScript is working and the element
+      // changes its role to `tab`.
+      this.setAttribute('role', 'tab');
+      if (!this.id) console.warn('All DashTabs must have an ID');
```

Comment:
I think you need to `return` here otherwise when you try to access `this.id` on 272 it'll be undefined